src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -237,7 +237,9 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             // not function properly.
             Href = output.Attributes[HrefAttributeName]?.Value as string;
 
-            var modeResult = AttributeMatcher.DetermineMode(context, ModeDetails);
+            var determinePartiallyMatchedAttributes = Logger.IsEnabled(
+                ModeMatchResultLoggerExtensions.PartiallyMatchedAttributesLogLevel);
+            var modeResult = AttributeMatcher.DetermineMode(context, ModeDetails, determinePartiallyMatchedAttributes);
             Logger.TagHelperModeMatchResult(modeResult, context.UniqueId, ViewContext.View.Path, this);
```

Comment:
Your logic makes me :cry: 

We had said we didn't want Tag Helpers to throw (they're not controls) as a general rule. We were going to use the logging system to raise usage warnings to the dev, but this is as far as we got (we had specced up a logging Tag Helper that would output the warnings to the console to the browser, but it never got done).

I'm OK to remove all this for now, but don't start throwing exceptions, just no-op.